src/Type/Php/ArraySliceFunctionReturnTypeExtension.php

```diff
@@ -25,49 +28,22 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
 	{
 		$args = $functionCall->getArgs();
-		if (count($args) < 1) {
-			return null;
-		}
-
-		$valueType = $scope->getType($args[0]->value);
-		if (!$valueType->isArray()->yes()) {
+		if (count($args) < 2) {
 			return null;
 		}
 
-		$offsetType = isset($args[1]) ? $scope->getType($args[1]->value) : null;
-		$limitType = isset($args[2]) ? $scope->getType($args[2]->value) : null;
-
-		$constantArrays = $valueType->getConstantArrays();
-		if (count($constantArrays) > 0) {
-			$preserveKeysType = isset($args[3]) ? $scope->getType($args[3]->value) : null;
-
-			$offset = $offsetType instanceof ConstantIntegerType ? $offsetType->getValue() : 0;
-			$limit = $limitType instanceof ConstantIntegerType ? $limitType->getValue() : null;
-			$preserveKeys = $preserveKeysType !== null && $preserveKeysType->isTrue()->yes();
-
-			$results = [];
-			foreach ($constantArrays as $constantArray) {
-				$results[] = $constantArray->slice($offset, $limit, $preserveKeys);
-			}
-
-			return TypeCombinator::union(...$results);
+		$arrayType = $scope->getType($args[0]->value);
+		if ($arrayType->isArray()->no()) {
+			return $this->phpVersion->arrayFunctionsReturnNullWithNonArray() ? new NullType() : new NeverType();
 		}
 
-		if ($valueType->isIterableAtLeastOnce()->yes()) {
-			$optionalOffsetsType = TypeCombinator::union($valueType, new ConstantArrayType([], []));
-
-			$zero = new ConstantIntegerType(0);
-			if (
-				($offsetType === null || $zero->isSuperTypeOf($offsetType)->yes())
-				&& ($limitType === null || IntegerRangeType::fromInterval(1, null)->isSuperTypeOf($limitType)->yes())
-			) {
-				return TypeCombinator::intersect($optionalOffsetsType, new NonEmptyArrayType());
-			}
+		$offsetType = $scope->getType($args[1]->value);
+		$lengthType = isset($args[2]) ? $scope->getType($args[2]->value) : new NullType();
 
-			return $optionalOffsetsType;
-		}
+		$preserveKeysType = isset($args[3]) ? $scope->getType($args[3]->value) : new ConstantBooleanType(false);
+		$preserveKeys = (new ConstantBooleanType(true))->isSuperTypeOf($preserveKeysType);
```

Comment:
maybe

```suggestion
		$preserveKeys = $preserveKeysType->isTrue()->yes();
```

Reply:
thx, good idea! slightly different but nice cleanup 2d50299d28611c61e3864b6f7863c5a6bdc35d29

(I'll adapt this same pattern I started sometime in the other places too in a dedicated MR :))